lib/hal_api/controller/actions.rb

```diff
@@ -72,7 +72,8 @@ def show_options
   def valid_params_for_action(action)
     (params.permit(*self.class.valid_params_list(action)) || {}).tap do |p|
       p[:zoom] = zoom_param if zoom_param
-    end
+    end.
+    to_h
```

Comment:
I can't imagine we lose much in converting to a hash, this looks nice and simple